File: Gradle/GradleTransformWithASMDemo/app/src/main/java/com/dream/gradletransformwithasmdemo/MainActivity.java
package com.dream.gradletransformwithasmdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.dream.customannotation.CostTime;

public class MainActivity extends AppCompatActivity {

    @CostTime
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

}

File: Protobuf/ProtobufDemo/app/src/main/java/com/dream/protobufdemo/MainActivity.kt
package com.dream.protobufdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

   //protoc -I=./app/src/main/proto --java_out=./app/src/main/java person.proto
   //protoc --java_out=./app/src/main/java ./app/src/main/proto/person.proto

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}